query/src/main/scala/filodb/query/exec/PromQlExec.scala

```diff
@@ -0,0 +1,117 @@
+package filodb.query.exec
+
+import com.softwaremill.sttp.akkahttp.AkkaHttpBackend
+import com.softwaremill.sttp.circe._
+import com.typesafe.scalalogging.StrictLogging
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.Observable
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import filodb.core.DatasetRef
+import filodb.core.metadata.Column.ColumnType
+import filodb.core.metadata.Dataset
+import filodb.core.query._
+import filodb.core.store.ChunkSource
+import filodb.memory.format.RowReader
+import filodb.memory.format.ZeroCopyUTF8String._
+import filodb.query._
+
+case class PromQlExec(id: String,
+                      dispatcher: PlanDispatcher,
+                      dataset: DatasetRef, params: PromQlQueryParams,
+                      submitTime: Long = System.currentTimeMillis()) extends LeafExecPlan {
+
+  protected def args: String = ""
+
+  /**
+    * Limit on number of samples returned by this ExecPlan
+    */
+  override def limit: Int = ???
+
+  /**
+    * Sub classes should override this method to provide a concrete
+    * implementation of the operation represented by this exec plan
+    * node
+    */
+  override protected def doExecute(source: ChunkSource, dataset: Dataset, queryConfig: QueryConfig)
+                                  (implicit sched: Scheduler, timeout: FiniteDuration): Observable[RangeVector] = ???
+
+  /**
+    * Sub classes should implement this with schema of RangeVectors returned
+    * from doExecute() abstract method.
+    */
+  override protected def schemaOfDoExecute(dataset: Dataset): ResultSchema = ???
+
+  val columns: Seq[ColumnInfo] = Seq(ColumnInfo("timestamp", ColumnType.LongColumn),
+    ColumnInfo("value", ColumnType.DoubleColumn))
+  val recSchema = SerializableRangeVector.toSchema(columns)
+  val builder = SerializableRangeVector.toBuilder(recSchema)
```

Comment:
The above need not be present for each ExecPlan object. 
Move it out to companion object.